packages/contracts-bedrock/src/cannon/libraries/MIPS64Instructions.sol

```diff
@@ -213,376 +241,386 @@ library MIPS64Instructions {
-                    return (_rs - _rt);
+                else if (_args.fun == 0x2e) {
+                    return (_args.rs - _args.rt);
                 }
                 // dsubu
-                else if (_fun == 0x2f) {
-                    return (_rs - _rt);
+                else if (_args.fun == 0x2f) {
+                    return (_args.rs - _args.rt);
                 }
                 // dsll
-                else if (_fun == 0x38) {
-                    return _rt << ((_insn >> 6) & 0x1f);
+                else if (_args.fun == 0x38) {
+                    return _args.rt << ((_args.insn >> 6) & 0x1f);
                 }
                 // dsrl
-                else if (_fun == 0x3A) {
-                    return _rt >> ((_insn >> 6) & 0x1f);
+                else if (_args.fun == 0x3A) {
+                    return _args.rt >> ((_args.insn >> 6) & 0x1f);
                 }
                 // dsra
-                else if (_fun == 0x3B) {
-                    return uint64(int64(_rt) >> ((_insn >> 6) & 0x1f));
+                else if (_args.fun == 0x3B) {
+                    return uint64(int64(_args.rt) >> ((_args.insn >> 6) & 0x1f));
                 }
                 // dsll32
-                else if (_fun == 0x3c) {
-                    return _rt << (((_insn >> 6) & 0x1f) + 32);
+                else if (_args.fun == 0x3c) {
+                    return _args.rt << (((_args.insn >> 6) & 0x1f) + 32);
                 }
                 // dsrl32
-                else if (_fun == 0x3e) {
-                    return _rt >> (((_insn >> 6) & 0x1f) + 32);
+                else if (_args.fun == 0x3e) {
+                    return _args.rt >> (((_args.insn >> 6) & 0x1f) + 32);
                 }
                 // dsra32
-                else if (_fun == 0x3f) {
-                    return uint64(int64(_rt) >> (((_insn >> 6) & 0x1f) + 32));
+                else if (_args.fun == 0x3f) {
+                    return uint64(int64(_args.rt) >> (((_args.insn >> 6) & 0x1f) + 32));
                 } else {
                     revert("MIPS64: invalid instruction");
                 }
             } else {
                 // SPECIAL2
-                if (_opcode == 0x1C) {
+                if (_args.opcode == 0x1C) {
                     // mul
-                    if (_fun == 0x2) {
-                        return signExtend(uint32(int32(uint32(_rs)) * int32(uint32(_rt))), 32);
+                    if (_args.fun == 0x2) {
+                        return signExtend(uint32(int32(uint32(_args.rs)) * int32(uint32(_args.rt))), 32);
                     }
                     // clz, clo
-                    else if (_fun == 0x20 || _fun == 0x21) {
-                        if (_fun == 0x20) {
-                            _rs = ~_rs;
+                    else if (_args.fun == 0x20 || _args.fun == 0x21) {
+                        if (_args.fun == 0x20) {
+                            _args.rs = ~_args.rs;
```

Comment:
(nit) if feasible, may be a little cleaner to set up a variable `rs = _args.rs` to avoid mutating the `_args` params for clz,clo,dclz,dclo.

Reply:
good idea. I'll do this for all inputs in `_args`.